pkg/pravega/pravega_controller.go

```diff
@@ -146,6 +149,12 @@ func makeControllerConfigMap(pravegaCluster *api.PravegaCluster) *corev1.ConfigM
 }
 
 func makeControllerService(pravegaCluster *api.PravegaCluster) *corev1.Service {
+
+	serviceType := corev1.ServiceTypeClusterIP
+	if pravegaCluster.Spec.ExternalAccess {
+		serviceType = corev1.ServiceTypeLoadBalancer
```

Comment:
Given that the use of an LB here is a workaround, as opposed to a NodePort, we should make it explicit so that we can use NodePort as the default approach in future.  e.g.
```
externalAccess:
  enabled: true
  type: LoadBalancer
```

Reply:
Nonetheless I strongly feel it should be an explicit option, not the default.

Reply:
I've added support for `NodePort` service type. However, I've set `LoadBalancer` as the default option because it's the only option that works in GKE and PKS at this moment.